src/index.ts

```diff
@@ -45,107 +143,64 @@ export default function cssInjectedByJsPlugin({
                 const htmlChunk = bundle[name] as OutputAsset;
-                let replacedHtml = htmlChunk.source as string;
+                let replacedHtml =
+                    htmlChunk.source instanceof Uint8Array
+                        ? new TextDecoder().decode(htmlChunk.source)
+                        : `${htmlChunk.source}`;
 
                 cssAssets.forEach((cssName) => {
                     replacedHtml = removeLinkStyleSheets(replacedHtml, cssName);
                     htmlChunk.source = replacedHtml;
                 });
             }
 
-            const jsAssetsFilter =
-                typeof jsAssetsFilterFunction == 'function' ? jsAssetsFilterFunction : defaultJsAssetsFilter;
-
-            let jsAssetTargets = [];
-            if (typeof jsAssetsFilterFunction != 'function') {
-                const jsAssets = Object.keys(bundle).filter(
-                    (i) => isJsOutputChunk(bundle[i]) && defaultJsAssetsFilter(bundle[i] as OutputChunk)
-                );
-
-                const jsTargetFileName = jsAssets[jsAssets.length - 1];
-                if (jsAssets.length > 1) {
-                    warnLog(
-                        `[vite-plugin-css-injected-by-js] has identified "${jsTargetFileName}" as one of the multiple output files marked as "entry" to put the CSS injection code. However, if this is not the intended file to add the CSS injection code, you can use the "jsAssetsFilterFunction" parameter to specify the desired output file (read docs).`
-                    );
-                    if (isDebug) {
-                        debugLog(
-                            `[vite-plugin-css-injected-by-js] identified js file targets: ${jsAssets.join(
-                                ', '
-                            )}. Selected "${jsTargetFileName}".\n`
-                        );
-                    }
-                }
-
-                // This should be always the root of the application
-                jsAssetTargets.push(bundle[jsTargetFileName] as OutputChunk);
-            } else {
-                const jsAssets = Object.keys(bundle).filter(
-                    (i) => isJsOutputChunk(bundle[i]) && jsAssetsFilter(bundle[i] as OutputChunk)
-                );
-
-                jsAssets.forEach((jsAssetKey) => {
-                    jsAssetTargets.push(bundle[jsAssetKey] as OutputChunk);
+            const buildCssCode = (cssToInject: string) =>
+                buildCSSInjectionCode({
+                    cssToInject: typeof preRenderCSSCode == 'function' ? preRenderCSSCode(cssToInject) : cssToInject,
+                    styleId,
+                    injectCode,
+                    injectCodeFunction,
+                    useStrictCSP,
                 });
+
+            if (relativeCSSInjection) {
+                const assetsWithCss = buildJsCssMap(bundle, jsAssetsFilterFunction);
+                await relativeCssInjection(bundle, assetsWithCss, buildCssCode);
```

Comment:
After we call `buildJsCssMap` we are sure that every piece of CSS is injected somewhere? And no file.css is generated yet?

Reply:
Yes it's better to low a warning